src/hooks/useDatabase/utils.ts

```diff
@@ -0,0 +1,20 @@
+import { Application, Collection, ModelBase } from '~/@types/common';
+
+export const mapEntry = <T extends ModelBase>(collection: Collection, entry: T): T => {
+  const mappedBase = {
+    ...entry,
+    createdAt: new Date(entry.createdAt),
```

Comment:
`entry.createdAt` is type of Date so why you wrap it again into another Date ?

Reply:
You are right! I simplified the fact that Local Storage stores dates as strings.
I pushed changes that add `DBModelBase` and `DBApplication` types. I had to change types a little bit in `useDatabase`.
Also fixed bug in `updateById` function - I mapped the entry twice (first time in `get` function and next in `updateById`) - thanks for spotting the issue with types!